frappe/model/base_document.py

```diff
@@ -105,9 +105,6 @@ def update(self, d):
 			})
 		"""
 
-		# QUESTION: why do we need the 1st for loop?
-		# we're essentially setting the values in d, in the 2nd for loop (?)
-
```

Comment:
can i please know the answer to this? the only reason i could think of is to keep those default fields first in ordering (?)

Reply:
not fully sure, it was added here: https://github.com/frappe/frappe/commit/445ba943c75280b094030210ed0fc44964e058b2

Reply:
well now we know 🤣

Reply:
the best way to know /s